### PR TITLE
feat: implement inline manifests in the machine configuration

### LINKED ISSUE #3416: inline bootstrap manifests
Add an option to the machine configuration like `cluster.inlineManifests` which allows multi-line strings as values, e.g.:

```yaml
cluster:
   inlineManifests:
     - name: foo-manifest
       manifest: | 
         metadata:
              name: foo
```

We need to update manifests controllers to parse the manifests and add them to the global list of manifests.

`name` should be unique across manifests (it will be used to identify resource holding the manifest, to give context to the errors if the manifest doesn't parse).

### PR DESCRIPTION
**Pull request by smira**
Inline manifests work exactly same way as extra manifests, but the
manifest itself can be stored in the config body.

Example config patch:

```
--config-patch '[{"op": "replace", "path": "/cluster/inlineManifests", "value": [{"name": "foo", "contents": "apiVersion: v1\nkind: Namespace\nmetadata:\n  name: ci\n"}]}]'
```

Fixes #3416 

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: internal/app/machined/pkg/controllers/k8s/extra_manifest.go [hunk 1]
```diff
@@ -139,8 +139,22 @@ func (ctrl *ExtraManifestController) Run(ctx context.Context, r controller.Runti
 	}
 }
 
-func (ctrl *ExtraManifestController) download(ctx context.Context, r controller.Runtime, logger *log.Logger, manifest config.ExtraManifest) (id resource.ID, err error) {
-	id = fmt.Sprintf("%s-%s", manifest.Priority, manifest.URL)
+func (ctrl *ExtraManifestController) process(ctx context.Context, r controller.Runtime, logger *log.Logger, manifest config.ExtraManifest) (id resource.ID, err error) {
```
Comment (AlekSi) [inline internal/app/machined/pkg/controllers/k8s/extra_manifest.go:142]: I think that long function should become shorter, not longer:

```
func process(…) {
  id := …

  if if manifest.InlineManifest != "" {
    return ctrl.processInline(…)
  }

  return ctrl.processURL(…)
}
```

Comment (smira) [reply internal/app/machined/pkg/controllers/k8s/extra_manifest.go:142]: thanks, fixed!

**Review by Unix4ever — APPROVED** [review]

**Comment by AlekSi** [comment]
@smira There is a merge conflict

**Comment by smira** [comment]
> @smira There is a merge conflict

I'll get it resolved, but it doesn't block the review process hopefully

**Review by AlekSi — APPROVED** [review]

**Pull request merged**